### PR TITLE
Change to LibraryImport

### PR DESCRIPTION
**Pull request by Quacktiamauct**
Change all uses of `DllImport` to `LibraryImport`

https://learn.microsoft.com/en-us/dotnet/standard/native-interop/pinvoke-source-generation

### CODE REVIEW

File: src/ProjectOrigin.PedersenCommitment/Ristretto/Generator.cs
Comment (MartinSchmidt) [inline src/ProjectOrigin.PedersenCommitment/Ristretto/Generator.cs:1]: Maybe we could move the library name `"rust_ffi"` to a constant in a seperate file, so we only reference it a single place with the string name, if one wants to later rename the packagee from rust_ffi  where I personally don't know what the `ffi` stands for

File: src/ProjectOrigin.PedersenCommitment/Ristretto/Point.cs [hunk 1]
```diff
@@ -5,47 +5,48 @@
...
+        [LibraryImport("rust_ffi", EntryPoint = "ristretto_point_from_uniform_bytes")]
+        internal static partial IntPtr FromUniformBytes(byte[] bytes);
 
         // TODO: check if byte[] is a sane argument
-        [DllImport("rust_ffi", EntryPoint = "ristretto_point_compress")]
-        internal static extern void Compress(IntPtr self, byte[] bytes_ptr);
+        [LibraryImport("rust_ffi", EntryPoint = "ristretto_point_compress")]
+        internal static partial void Compress(IntPtr self, byte[] bytes_ptr);
 
         // TODO: check if byte[] is a sane argument
```
Comment (MartinSchmidt) [inline src/ProjectOrigin.PedersenCommitment/Ristretto/Point.cs:19]: could you remove the todo?

**Review by MartinSchmidt — CHANGES_REQUESTED** [review]
Maybe we could move the library name "rust_ffi" to a constant in a seperate file, so we only reference it a single place with the string name, if one wants to later rename the package from rust_ffi where I personally don't know what the ffi stands for?

**Comment by Quacktiamauct** [comment]
FFI stands for [foreign function interface](https://en.wikipedia.org/wiki/Foreign_function_interface). I am not sure if the name can be separated out into a separate file. It is not something I have seen before, although I get the incentive. Maybe the string can be defined in the `.csproj` file and then replaced compile-time by the build system?

**Comment by Quacktiamauct** [comment]
Nvm, managed something. It is now defined in the `Util.cs` file, although it could maybe be moved more out.

File: src/ProjectOrigin.PedersenCommitment/Ristretto/Util.cs [hunk 2]
```diff
@@ -19,7 +21,7 @@ internal static byte[] ToByteArray(this ulong value, uint arrayLength)
         return outputArray;
     }
 
-    [LibraryImport("rust_ffi", EntryPoint = "fill_bytes")]
+    [LibraryImport(LIBRARY, EntryPoint = "fill_bytes")]
     internal static partial void FillBytes(RawVec raw, byte[] dst);
 
     [LibraryImport("rust_ffi", EntryPoint = "free_vec")]
```
Comment (MartinSchmidt) [inline src/ProjectOrigin.PedersenCommitment/Ristretto/Util.cs:27]: missed one

**Review by MartinSchmidt — CHANGES_REQUESTED** [review]
@Quacktiamauct  otherwise good :)

**Review by MartinSchmidt — APPROVED** [review]

**Pull request merged**